type_description_interfaces/README.md

```diff
@@ -0,0 +1,22 @@
+# type_description_interfaces
+
+A package containing message and service definitions for describing and communicating descriptions of other types.
+
+## Messages (.msg)
+
+* [Field](msg/Field.msg): Represents a single field in a ROS Interface Type
+* [FieldType](msg/FieldType.msg): Represents the type of a single field in a ROS Interface Type
+* [IndividualTypeDescription](msg/IndividualTypeDescription.msg): Represents a single ROS Message Type by itself
+* [TypeDescription](msg/TypeDescription.msg): Represents a single ROS Message Type along with all referenced types
+
+## Quality Declaration
+
+This package claims to be in the **Quality Level 1** category, see the [Quality Declaration](QUALITY_DECLARATION.md) for more details.
+
+## TODO
+
+* How to store default values, should store default values? Current storing as string, but will impact hashing.
+* Do we (will we) support fixed sized strings vs. just having bounded strings?
+* Should we pick limits (bounds) for strings in these types, e.g. type names, default values, etc?
+* Should we pick limits (bounds) for sequences in these types, e.g. number of referenced types?
+* How should we handle Services and Actions?
```

Comment:
As I'm implementing hashing, I am running into this.

Couple of options I've been thinking of (using Service for example, but applies to Action as well):
1. Only represent Messages, so compatibility/hashing is only on MyServiceRequest, MyServiceResponse, there isn't the higher level matching. This could allow request to be compatible but response to go out of sync, maybe ill advised because of that.
2. Define an `IndividualTypeDescription` for a service, that has just two fields that are referenced types for the implementing messages, something like:

```
type_description:
  type_name: MyService
  fields:
    - name: request
      type:
        type_id: 0 
        nested_type_id: MyServiceRequest
    - name: response
      type:
        type_id: 0
        nested_type_id: MyServiceResponse
referenced_type_descriptions:
  - type_name: MyServiceRequest
    fields:
... etc
```

That second one seems maybe better for real use cases, and it allows us to have a single Type Version Hash at the top level of the interface, so this could possibly be defined in e.g. `idl__struct.h` rather than propagating through e.g. `srv__struct.h` to `msg__struct.h`

Reply:
This seems totally reasonable to me. I don't think we need changes in this pr for this approach, so I'll just give it a 👍  and resolve this conversation. Thanks!